### PR TITLE
Ensure terminal doesn't render partial lines & Snap terminal scrolling to line height

### PR DESCRIPTION
**Pull request by levkropp**
Via a fork of the flutter xterm.dart plugin, editing ui/render.dart

### CODE REVIEW

**Comment by andrei-toterman** [comment]
Nice, the terminals scrolls nicely and there are no partial lines. One issue though is that now the terminal does not autoscroll anymore when new lines appear. For example, if I do `sudo apt update && sudo apt upgrade` there are many new lines written to the terminal, but I am not automatically taken to the bottom of them as they appear.

**Review by andrei-toterman — APPROVED** [review]
It works really nicely now :smile: Thanks, @levkropp!

**Comment by ricab** [comment]
Oh, I meant to secondary-review this...

File: src/client/gui/pubspec.yaml [hunk 1]
```diff
@@ -52,7 +52,10 @@ dependencies:
       url: https://github.com/google/flutter-desktop-embedding.git
       path: plugins/window_size
       ref: 6c66ad2
-  xterm: ^4.0.0
+  xterm:
+    git:
+      url: https://github.com/levkropp/xterm.dart
```
Comment (ricab) [inline src/client/gui/pubspec.yaml:57]: This should live in the Canonical org, but I see there are similar cases here. I'll propose fixing all that on an upcoming pulse.

**Review by ricab — COMMENTED** [review]
Late approval :heavy_check_mark:

**Pull request merged**